cpp/log/cert.cc

```diff
@@ -883,7 +914,8 @@ Cert::Status Cert::IsValidNameConstrainedIntermediateCa() const {
 
   // If it's not a CA cert or there is no name constraint extension then we
   // don't need to apply the rules any further
-  if (HasBasicConstraintCATrue() != Cert::TRUE
+  StatusOr<bool> has_ca_constraint_status = HasBasicConstraintCATrue();
+  if (StatusOrBoolToCertStatus(has_ca_constraint_status) != Cert::TRUE
```

Comment:
Just trying to preserve the structure of the logic for the intermediate state so it reads similarly. I can update it now but thought it would be easier to review.